### ISSUE TITLE
TakeSnapshot image is unrecognizable

### ISSUE THREAD

**Issue by jackbayliss** [cordova-plugin-camera-preview/cordova-plugin-camera-preview#537]
Using the takeSnapshot function #525, I'm getting a Base64 image that is unrecognizable. See the photos below, I used takeSnapshot, then takeImage. Not sure if takeSnapchat is supposed to give a result like this?

[see images](https://imgur.com/a/gb6g4g3) 

This was taken using a Samsung J4+ for reference.

@westonganger

**Comment by jackbayliss** [comment]
@huxia Not sure if this is expected behaviour?

**Comment by huxia** [comment]
I don't have the "Samsung J4+" device, looking at the example, I guess the bug is during the NV21 image processing.

could you please help adding a break point here: https://github.com/cordova-plugin-camera-preview/cordova-plugin-camera-preview/blob/master/src/android/CameraActivity.java#L600 and share the contents of "bytes" byte array if possible?

many thanks!

**Comment by huxia** [comment]
```java
  public void takeSnapshot(final int quality) {
    mCamera.setPreviewCallback(new Camera.PreviewCallback() {
      @Override
      public void onPreviewFrame(byte[] bytes, Camera camera) {
        try {
          // BREAKING POINT AT BELOW LINE, so we can capture the "bytes" data
          Camera.Parameters parameters = camera.getParameters(); 
          Camera.Size size = parameters.getPreviewSize();
```

**Comment by huxia** [comment]
@jackbayliss it would be great if you can share more variable like `mPreview.getDisplayOrientation()` and `mPreview.getCameraFacing()``parameters.getPreviewSize()`

**Comment by jackbayliss** [comment]
@huxia  Bare with me, I'll try and get this as soon as I can.

**Comment by wjeb** [comment]
Have same problem. How can i help(i have device + phonegap builder) ?

**Comment by wjeb** [comment]
That problem is because after rotating wasn`t switched width/height. Here is my solution:
`          Camera.Parameters parameters = camera.getParameters();
          Camera.Size size = parameters.getPreviewSize();
          int orientation = mPreview.getDisplayOrientation();
		  
		  int newWidth = size.width;
		  int newHeight = size.height;
		  
          if (mPreview.getCameraFacing() == Camera.CameraInfo.CAMERA_FACING_FRONT) {
            switch (orientation) {
              case 90:
				newWidth = size.height;
				newHeight = size.width;
                bytes = rotateNV21(bytes, size.width, size.height, 270);
                break;
              case 270:
				newWidth = size.height;
				newHeight = size.width;
                bytes = rotateNV21(bytes, size.width, size.height, 90);
                break;
              default:
                bytes = rotateNV21(bytes, size.width, size.height, orientation);
                break;
            }
          } else {
			newWidth = size.height;
			newHeight = size.width;
            bytes = rotateNV21(bytes, size.width, size.height, 90);
          }
		  
          YuvImage yuvImage = new YuvImage(bytes, parameters.getPreviewFormat(), newWidth, newHeight, null);
          ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();

          Rect rect = new Rect(0, 0, newWidth, newHeight);`

**Comment by huxia** [comment]
Thanks @wjeb for find out the root cause, I've done the fix & tested with my phone.

**Comment by westonganger** [comment]
Fixed in master via #542
